packages/mui-joy/src/FormControl/FormControl.test.tsx

```diff
@@ -110,6 +110,24 @@ describe('<FormControl />', () => {
 
       expect(getByRole('textbox')).to.have.attribute('required');
     });
+
+    it('should inherit htmlFor from FormControl if htmlFor is undefined', () => {
+      const { container } = render(
+        <FormControl>
+          <FormLabel htmlFor={undefined} />
+        </FormControl>,
+      );
+      expect(container.firstChild?.firstChild).to.have.attribute('for');
```

Comment:
Let's use `getByLabelText` query. It is recommended over `container` queries in React testing library.
```suggestion
      expect(getByLabelText('label')).to.have.attribute('for');
```

Reply:
I've used `getByText` because `getByLabelText` gets input instead of the label itself